pettingzoo/mpe/scenarios/group_spread.py

```diff
@@ -48,44 +47,37 @@ def make_world(self, groups, num_colors=None, reward_per_group=True, randomise_a
 
     def reset_world(self, world, np_random):
         if self.randomise_all_colors:
-            # generate colors if they are randomised for each episode
-            self.colors = [np.random.random(3) for _ in self.groups]
-            color_idxs = np.arange(len(self.groups))
-        else:
-            # assign one of the available colors to a group
-            color_idxs = np.random.choice(self.num_colors, len(self.groups), replace=False)
-
-        # random properties for agents
+            # TODO: do nicer to avoid overlap here?
+            # randomly generate new colors for each episode
+            self.colors = [np.random.random(3) for _ in range(self.num_colors)]
+
+        # assign one of the available colors to a group
+        color_idxs = np.random.choice(self.num_colors, len(self.groups), replace=False)
+
+        # assign group / color properties to agents
         for i, agent in zip(self.group_indices, world.agents):
             agent.color = self.colors[color_idxs[i]]
             agent.color_idx = color_idxs[i]
             agent.group = i
 
-        # random properties for landmarks
-        if self.shuffle_obs_per_agent:
-            for i, landmark in zip(self.group_indices, world.landmarks):
-                landmark.color = self.colors[color_idxs[i]]
-                landmark.color_idx = color_idxs[i]
-                landmark.group = i
-            # assign each agent a landmark and agent ordering that wil determine order in the agent obs
-            # this allows entities to be shuffled per agent while keeping consistent for the episode
-            for agent in world.agents:
-                landmark_obs_order = np.arange(len(world.landmarks))
-                random.shuffle(landmark_obs_order)
-                agent.landmark_obs_order = landmark_obs_order
-                agent_obs_order = np.arange(len(world.agents))
-                random.shuffle(agent_obs_order)
-                agent.agent_obs_order = agent_obs_order
-        else:
-            # assign landmark groups randomly as landmarks won't be shuffled in obs
-            landmarks_ids = np.arange(len(world.landmarks))
-            np_random.shuffle(landmarks_ids)
-            for group_id, land_id in zip(self.group_indices, landmarks_ids):
-                world.landmarks[land_id].color = self.colors[color_idxs[group_id]]
-                world.landmarks[land_id].color_idx = color_idxs[group_id]
-                world.landmarks[land_id].group = group_id
-
-        # set random initial states
+        # assign properties to landmarks
+        landmarks_ids = np.random.choice(self.num_landmarks, self.num_landmarks, replace=False)
+        for group_id, land_id in zip(self.group_indices, landmarks_ids):
+            world.landmarks[land_id].color = self.colors[color_idxs[group_id]]
+            world.landmarks[land_id].color_idx = color_idxs[group_id]
+            world.landmarks[land_id].group = group_id
```

Comment:
No need to differentiate this depending on whether the observations are randomly shuffled or not? If they are randomly shuffled, then now they are shuffled twice (assignment of groups to landmarks here is random + again for obs construction) and if not randomly shuffled obs-wise then they are still shuffled here as before.

Reply:
Actually now with the changes they are shuffled twice when `shuffle_obs_per_agent=True` since they are shuffled here and again when agent order is assigned. Previously they were only shuffled here if `shuffle_obs_per_agent=False` since they won't get shuffled again later. Although I don't think it really makes much difference and I agree this is much tidier now anyway.

Reply:
Yes, exactly. Previously, this ensured that everything is only shuffled once whereas now it can be shuffled twice but that shouldn't affect anything (twice uniform shuffling should be same as once) and code becomes simpler to read,